--- wireshark-1.11.3/configure.ac.orig	2014-06-05 12:40:02.363486127 +0100
+++ wireshark-1.11.3/configure.ac	2014-06-05 12:40:42.071212871 +0100
@@ -2575,8 +2575,8 @@
 #endif
 }], [AC_MSG_RESULT(ok);
 have_inet_pton=yes], [AC_MSG_RESULT(broken);
-have_inet_pton=no], [AC_MSG_RESULT(cross compiling, assume it is broken);
-have_inet_pton=no])],
+have_inet_pton=no], [AC_MSG_RESULT(cross compiling, assume it is ok);
+have_inet_pton=yes])],
 have_inet_pton=no)
 if test "$have_inet_pton" = no; then
   INET_PTON_LO="inet_pton.lo"
