--- ./doc/Makefile.am.orig	2011-09-08 01:10:15.000000000 +0100
+++ ./doc/Makefile.am	2014-07-07 16:51:10.636847348 +0100
@@ -130,24 +130,6 @@
 	--noindex							\
 	$(srcdir)/tshark.pod > tshark.html
 
-wireshark-filter.4: wireshark-filter.pod ../config.h
-	$(POD2MAN)					\
-	--section=4					\
-	--center="The Wireshark Network Analyzer"	\
-	--release=$(VERSION)				\
-	wireshark-filter.pod > wireshark-filter.4
-
-wireshark-filter.html: wireshark-filter.pod ../config.h $(top_srcdir)/docbook/ws.css
-	$(POD2HTML)								\
-	--title="wireshark-filter - The Wireshark Network Analyzer $(VERSION)"	\
-	--css=$(top_srcdir)/docbook/ws.css					\
-	--noindex								\
-	wireshark-filter.pod > wireshark-filter.html
-
-wireshark-filter.pod: wireshark-filter.pod.template  ../tshark
-	WIRESHARK_RUN_FROM_BUILD_DIRECTORY=1 ../tshark -G fields | \
-	$(PERL) $(srcdir)/dfilter2pod.pl $(srcdir)/wireshark-filter.pod.template > wireshark-filter.pod
-
 capinfos.1: capinfos.pod ../config.h
 	$(POD2MAN)					\
 	--center="The Wireshark Network Analyzer"	\
@@ -266,10 +248,7 @@
 	$(srcdir)/randpkt.pod > randpkt.html
 
 CLEANFILES =		\
-	wireshark-filter.pod \
 	wireshark.pod \
-	wireshark-filter.4 \
-	wireshark-filter.html \
 	wireshark.1	\
 	wireshark.html	\
 	capinfos.1	\
