--- postgresql/src/interfaces/ecpg/Makefile.orig	2014-09-26 13:40:23.556716871 +0100
+++ postgresql/src/interfaces/ecpg/Makefile	2014-09-26 13:40:30.156941616 +0100
@@ -27,4 +27,4 @@ clean distclean maintainer-clean:
 	$(MAKE) -C test clean
 
 check checktcp installcheck: all
-	$(MAKE) -C test $@
+#	$(MAKE) -C test $@
--- postgresql/src/Makefile.orig	2014-09-26 13:42:47.019199129 +0100
+++ postgresql/src/Makefile	2014-09-26 13:43:13.207165040 +0100
@@ -25,7 +25,7 @@ SUBDIRS = \
 	bin \
 	pl \
 	makefiles \
-	test/regress
+#	test/regress
 
 # There are too many interdependencies between the subdirectories, so
 # don't attempt parallel make here.
